mailbox/jpa/src/test/java/org/apache/james/mailbox/jpa/mail/task/JPARecomputeCurrentQuotasServiceTest.java

```diff
@@ -0,0 +1,128 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.jpa.mail.task;
+
+import javax.persistence.EntityManagerFactory;
+
+import org.apache.commons.configuration2.BaseHierarchicalConfiguration;
+import org.apache.james.backends.jpa.JpaTestCluster;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.jpa.JPAMailboxFixture;
+import org.apache.james.mailbox.jpa.JPAMailboxSessionMapperFactory;
+import org.apache.james.mailbox.jpa.JpaMailboxManagerProvider;
+import org.apache.james.mailbox.jpa.mail.JPAModSeqProvider;
+import org.apache.james.mailbox.jpa.mail.JPAUidProvider;
+import org.apache.james.mailbox.jpa.quota.JpaCurrentQuotaManager;
+import org.apache.james.mailbox.quota.CurrentQuotaManager;
+import org.apache.james.mailbox.quota.UserQuotaRootResolver;
+import org.apache.james.mailbox.store.StoreMailboxManager;
+import org.apache.james.mailbox.store.mail.task.RecomputeCurrentQuotasService;
+import org.apache.james.mailbox.store.mail.task.RecomputeCurrentQuotasServiceContract;
+import org.apache.james.mailbox.store.quota.CurrentQuotaCalculator;
+import org.apache.james.mailbox.store.quota.DefaultUserQuotaRootResolver;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.jpa.JPAUsersRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+
+import com.google.common.collect.ImmutableList;
+
+@Disabled("JAMES-3138 Broken...")
```

Comment:
Still broken? Can we get a better comment?

Reply:
Yes still... I'm not too comfortable with JPA. I suspect something to do with the usersRepository or QuotaRootResolver but I'm not sure (it's an initialization issue)

What do you suggest as a comment? Been twice I changed this already I think

Reply:
> (it's an initialization issue)

Do we have user definition in `mailbox/jpa/src/main/resources/META-INF/persistence.xml` ?

Take away: be precise in the comment disabling things, others might have some clues ;-)

Reply:
Alright let me recheck, thanks !

Reply:
It looks like a persistence issue but I really cannot wrap my head around it... if it gives you any ideas, here are some logs:

```
[ERROR] recomputeCurrentQuotasShouldNotUpdateContextWhenNoData  Time elapsed: 0.005 s  <<< ERROR!
org.apache.openjpa.persistence.ArgumentException: There is no query with the name "listUserNames" defined for any of the known persistent classes: [org.apache.james.mailbox.jpa.quota.m$
del.MaxGlobalMessageCount, org.apache.james.mailbox.jpa.mail.model.JPAUserFlag, org.apache.james.mailbox.jpa.quota.model.MaxUserStorage, org.apache.james.mailbox.jpa.mail.model.openjpa$
AbstractJPAMailboxMessage, org.apache.james.mailbox.jpa.quota.model.JpaCurrentQuota, org.apache.james.mailbox.jpa.mail.model.JPAMailboxAnnotation, org.apache.james.mailbox.jpa.user.mod$
l.JPASubscription, org.apache.james.mailbox.jpa.quota.model.MaxGlobalStorage, org.apache.james.mailbox.jpa.quota.model.MaxDomainMessageCount, org.apache.james.mailbox.jpa.mail.model.JP$
Mailbox, org.apache.james.mailbox.jpa.quota.model.MaxDomainStorage, org.apache.james.mailbox.jpa.quota.model.MaxUserMessageCount, org.apache.james.mailbox.jpa.mail.model.JPAProperty, o$
g.apache.james.mailbox.jpa.mail.model.openjpa.JPAMailboxMessage].

[ERROR] recomputeCurrentQuotasShouldUpdateContextWhenIncorrectQuotas  Time elapsed: 0.005 s  <<< ERROR!
org.apache.openjpa.persistence.ArgumentException: There is no query with the name "containsUser" defined for any of the known persistent classes: [org.apache.james.mailbox.jpa.quota.mo$
el.MaxGlobalMessageCount, org.apache.james.mailbox.jpa.mail.model.JPAUserFlag, org.apache.james.mailbox.jpa.quota.model.MaxUserStorage, org.apache.james.mailbox.jpa.mail.model.openjpa.$
bstractJPAMailboxMessage, org.apache.james.mailbox.jpa.quota.model.JpaCurrentQuota, org.apache.james.mailbox.jpa.mail.model.JPAMailboxAnnotation, org.apache.james.mailbox.jpa.user.mode$
.JPASubscription, org.apache.james.mailbox.jpa.quota.model.MaxGlobalStorage, org.apache.james.mailbox.jpa.quota.model.MaxDomainMessageCount, org.apache.james.mailbox.jpa.mail.model.JPA$
ailbox, org.apache.james.mailbox.jpa.quota.model.MaxDomainStorage, org.apache.james.mailbox.jpa.quota.model.MaxUserMessageCount, org.apache.james.mailbox.jpa.mail.model.JPAProperty, org
.apache.james.mailbox.jpa.mail.model.openjpa.JPAMailboxMessage].
```

Reply:
You miss JPAUser in the persistance class.